src/forms/list-alert.ts

```diff
@@ -1,34 +1,70 @@
-import { Account, Alert, AppCallRequest, ListAlertParams, ResponseResultWithData } from '../types';
+import { Account, Alert, AlertStatus, AppActingUser, AppCallRequest, ListAlertParams, Teams } from '../types';
 import { OpsGenieClient, OpsGenieOptions } from '../clients/opsgenie';
-import { ConfigStoreProps, KVStoreClient, KVStoreOptions } from '../clients/kvstore';
-import { ExceptionType, StoreKeys } from '../constant';
+import { AppsOpsGenie, ExceptionType, Routes } from '../constant';
 import { configureI18n } from '../utils/translations';
-import { tryPromise } from '../utils/utils';
+import { isUserSystemAdmin, replace, tryPromise } from '../utils/utils';
+import { allowMemberAction, getOpsGenieAPIKey, validateUserAccess } from '../utils/user-mapping';
+import { h6, hyperlink, joinLines, strikethrough } from '../utils/markdown';
+import { Exception } from '../utils/exception';
 
-export async function getAllAlertCall(call: AppCallRequest): Promise<[Alert[], Account]> {
-    const mattermostUrl: string | undefined = call.context.mattermost_site_url;
-    const botAccessToken: string | undefined = call.context.bot_access_token;
-    const i18nObj = configureI18n(call.context);
-
-    const options: KVStoreOptions = {
-        mattermostUrl: <string>mattermostUrl,
-        accessToken: <string>botAccessToken,
-    };
-    const kvStoreClient = new KVStoreClient(options);
+import { getAllTeamsCall } from './list-team';
 
-    const config: ConfigStoreProps = await kvStoreClient.kvGet(StoreKeys.config);
+export async function getAllAlertCall(call: AppCallRequest): Promise<string> {
+    const actingUser: AppActingUser | undefined = call.context.acting_user;
+    const isSystemAdmin: boolean = isUserSystemAdmin(actingUser);
+    const allowMember: boolean = allowMemberAction(call.context);
+    const i18nObj = configureI18n(call.context);
+    const apiKey = getOpsGenieAPIKey(call);
 
     const optionsOpsgenie: OpsGenieOptions = {
-        api_key: config.opsgenie_apikey,
+        api_key: apiKey,
     };
     const opsGenieClient = new OpsGenieClient(optionsOpsgenie);
 
     const params: ListAlertParams = {
         limit: 100,
     };
-    const teams: ResponseResultWithData<Alert[]> = await tryPromise(opsGenieClient.listAlert(params), ExceptionType.MARKDOWN, i18nObj.__('forms.error'));
+    let alerts: Alert[] = await tryPromise<Alert[]>(opsGenieClient.listAlert(params), ExceptionType.MARKDOWN, i18nObj.__('forms.error'));
+
+    if (!allowMember) {
+        throw new Exception(ExceptionType.MARKDOWN, i18nObj.__('general.validation-user.genie-action-invalid'));
+    }
+
+    if (!isSystemAdmin) {
+        const teams: Teams[] = await getAllTeamsCall(call);
+        const teamsIds: string[] = teams.map((team) => team.id);
+        alerts = alerts.filter((alert: Alert) => teamsIds.includes(alert.ownerTeamId));
```

Comment:
Are we able to provide the team IDs in the request to OpsGenie, rather than manually filtering them out here?

Reply:
Currently OpsGenie query alert can only filter by a team at time [(Search queries for alerts)](https://support.atlassian.com/opsgenie/docs/search-queries-for-alerts/).

teams : team1 | Name of the team.
-- | --

We tried to add more, but the response array is returned as empty